src/MSBuild/XMake.cs

```diff
@@ -2516,6 +2516,11 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
 #endif
 
             bool useTerminalLogger = ProcessTerminalLoggerConfiguration(commandLineSwitches, out string aggregatedTerminalLoggerParameters);
+
+            // This is temporary until we can remove the need for the environment variable.
+            // DO NOT use this environment variable for any new features as it will be removed without further notice.
+            Environment.SetEnvironmentVariable("_MSBUILDTLENABLED", useTerminalLogger ? "1" : "0");
```

Comment:
Just a curiosity - why is this underscore prefixed? We have other 'not supposed to be used' env vars - all starts with 'MSBUILD'

Reply:
It felt right to do it as we are doing it for properties and additionally, this variable should be removed in the future :) Let me know if you insist to remove it.

Reply:
Feels right to me too.

Reply:
Oh no - I was just curious.
Let's get this in